pkg/plugins/aws/ami/main.go

```diff
@@ -0,0 +1,107 @@
+package ami
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+)
+
+// https://docs.aws.amazon.com/sdk-for-go/api/service/ec2/#EC2.DescribeImages
+
+// AMI contains information to manipuliate AWS AMI information
+type AMI struct {
+	AccessKey  string // AWs access key
+	SecretKey  string // AWS secret key
+	Filters    Filters
+	Region     string
+	Endpoint   string
+	DryRun     bool
+	ec2Filters []*ec2.Filter
+}
+
+// Filter represents the updatecli configuration to define AMI filter
+// This datatype need to be convert the ec2.Filter.
+type Filter struct {
+	Name   string
+	Values string
+}
```

Comment:
you might want to define 2 types:
- amiSpec, a private type used to represent the "data" parsed from yaml.
  - this type should manage the yaml parsing and the validation of the data (raising error when there are semantic errors such as valid yaml but missing or empty directives)
- AMI, the top level type that has a private object `spec` of type `amispec`. This object would expose the public methode such as `Condition` , etc. and of course should have the private attributes related to AWS api such as `ec2Filters`.
  - The `Init` method (I like to use `New)  would be changed to have a spec object (from the YAML parsing) and would respond a fully initialized `AWS` object.

Reply:
I love this suggestion very much

Reply:
Implemented